app/common/src/main/java/com/fsck/k9/backends/AndroidAlarmManager.kt

```diff
@@ -62,12 +63,22 @@ class AndroidAlarmManager(
 
     override fun setAlarm(triggerTime: Long, callback: Callback) {
         this.callback.set(callback)
-        alarmManager.scheduleAlarm(triggerTime, pendingIntent)
+
+        if (AlarmManagerCompat.canScheduleExactAlarms(alarmManager)) {
+            AlarmManagerCompat.setExactAndAllowWhileIdle(
+                alarmManager,
+                AlarmManager.ELAPSED_REALTIME_WAKEUP,
+                triggerTime,
+                pendingIntent,
+            )
+        } else {
+            alarmManager.set(AlarmManager.ELAPSED_REALTIME_WAKEUP, triggerTime, pendingIntent)
```

Comment:
I added this as the old `AlarmManagerCompat` behaviour was different to the new compat version. We used to call `alarmManager.set()` when the API is below M, but the Android version calls `alarmManager.setExact()`. Not sure if it was intentionally like this.

Reply:
The condition would have to be `if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.M) …` then. And if we do that, there would be little reason to use AndroidX's `AlarmManagerCompat`. 

But using `setExact()` instead of `set()` is probably better. So I'd still go with my suggestion from above.

Reply:
The version we used is similar to `AlarmManagercompat.setAndAllowWhileIdle`, so if we would stick to exact alarms then I guess that there was a mixup.

Reply:
Good, I get rid of the old faulty implementation and go with the new compat version.